# 
# libmp4v2_narrowing.patch
#
# Reduce LC_NONE by a power of 2 to prevent LongCode promotion to unsigned int
# causing "narrowing" errors in recent GCC. The value is still sufficiently high
# as not to clash with any character values.
#

diff -Nurp libmp4v2-20191108_9084868.orig/libutil/Utility.h libmp4v2-20191108_9084868/libutil/Utility.h
--- libmp4v2-20191108_9084868.orig/libutil/Utility.h	2019-11-08 02:03:40.000000000 +0000
+++ libmp4v2-20191108_9084868/libutil/Utility.h	2021-08-19 23:40:34.724030480 +0100
@@ -52,7 +52,7 @@ class MP4V2_EXPORT Utility
 {
 protected:
     enum LongCode {
-        LC_NONE = 0xf0000000, // safe (cannot conflict with char values)
+        LC_NONE = 0x0f000000, // safe (cannot conflict with char values)
         LC_DEBUG,
         LC_VERBOSE,
         LC_HELP,
